packages/edit-post/src/plugins/index.js

```diff
@@ -19,6 +21,12 @@ registerPlugin( 'edit-post', {
 	render() {
 		return (
 			<>
+				<FilteredComponent
+					hookName="editor.MediaUpload"
+					namespace="core/edit-post/replace-media-upload"
+				>
+					{ () => <MediaUpload /> }
+				</FilteredComponent>
```

Comment:
The following would work as well but it wouldn't exactly translate from the previous implementation because it would take `props` if they are passed to the original component:
```suggestion
				<FilteredComponent
					hookName="editor.MediaUpload"
					namespace="core/edit-post/replace-media-upload"
					as={ MediaUpload }
				/>
```